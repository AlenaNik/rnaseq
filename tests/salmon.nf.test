nextflow_pipeline {

    name "Test pipeline with Salmon, skipping both QC and alignment"
    script "../main.nf"

    test("Params: --pseudo_aligner salmon --skip_qc --skip_alignment") {

        when {
            params {
                outdir = "$outputDir"
                pseudo_aligner = 'salmon'
                skip_qc = true
                skip_alignment = true
            }
        }

        then {
            def file_complement = UTILS.getAllFilesFromDir(params.outdir, null, UTILS.exclusionRegexesForUnstableFileNames)
            def snapshottable_files = UTILS.getAllFilesFromDir(params.outdir, UTILS.snapshottablePatterns, UTILS.exclusionRegexesForUnstableFileContents)

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnaseq_software_mqc_versions.yml"),
                    file_complement.collect { it.getName() },
                    snapshottable_files
                ).match() }
            )
        }
    }

    test("Params: --pseudo_aligner salmon --skip_qc --skip_alignment - stub") {

        options "-stub"

        when {
            params {
                outdir = "$outputDir"
                pseudo_aligner = 'salmon'
                skip_qc = true
                skip_alignment = true
            }
        }

        then {
            def file_complement = UTILS.getAllFilesFromDir(params.outdir, null, UTILS.exclusionRegexesForUnstableFileNames)
            def snapshottable_files = UTILS.getAllFilesFromDir(params.outdir, UTILS.snapshottablePatterns, UTILS.exclusionRegexesForUnstableFileContents)

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnaseq_software_mqc_versions.yml"),
                    file_complement.collect { it.getName() },
                    snapshottable_files
                ).match() }
            )
        }
    }
}
