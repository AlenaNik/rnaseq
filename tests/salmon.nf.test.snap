{
    "Params: --pseudo_aligner salmon --skip_qc --skip_alignment": {
        "content": [
            {
                "BBMAP_BBSPLIT": {
                    "bbmap": 39.01
                },
                "CAT_FASTQ": {
                    "cat": 8.3
                },
                "CUSTOM_CATADDITIONALFASTA": {
                    "python": "3.9.5"
                },
                "CUSTOM_GETCHROMSIZES": {
                    "getchromsizes": 1.2
                },
                "CUSTOM_TX2GENE": {
                    "python": "3.9.5"
                },
                "FQ_SUBSAMPLE": {
                    "fq": "0.9.1 (2022-02-22)"
                },
                "GTF2BED": {
                    "perl": "5.26.2"
                },
                "GTF_FILTER": {
                    "python": "3.9.5"
                },
                "GUNZIP_ADDITIONAL_FASTA": {
                    "gunzip": 1.1
                },
                "GUNZIP_GTF": {
                    "gunzip": 1.1
                },
                "SALMON_QUANT": {
                    "salmon": "1.10.1"
                },
                "SE_GENE": {
                    "bioconductor-summarizedexperiment": "1.32.0"
                },
                "TRIMGALORE": {
                    "trimgalore": "0.6.7",
                    "cutadapt": 3.4
                },
                "TXIMETA_TXIMPORT": {
                    "bioconductor-tximeta": "1.20.1"
                },
                "UNTAR_SALMON_INDEX": {
                    "untar": 1.34
                },
                "Workflow": {
                    "nf-core/rnaseq": "v3.16.0dev"
                }
            },
            [
                "RAP1_IAA_30M_REP1",
                "RAP1_IAA_30M_REP1.stats.txt",
                "RAP1_IAA_30M_REP1_trimmed_1.fastq.gz_trimming_report.txt",
                "RAP1_IAA_30M_REP1_trimmed_1_val_1_fastqc.html",
                "RAP1_IAA_30M_REP1_trimmed_1_val_1_fastqc.zip",
                "RAP1_IAA_30M_REP1_trimmed_2.fastq.gz_trimming_report.txt",
                "RAP1_IAA_30M_REP1_trimmed_2_val_2_fastqc.html",
                "RAP1_IAA_30M_REP1_trimmed_2_val_2_fastqc.zip",
                "RAP1_UNINDUCED_REP1",
                "RAP1_UNINDUCED_REP1.stats.txt",
                "RAP1_UNINDUCED_REP1_trimmed.fastq.gz_trimming_report.txt",
                "RAP1_UNINDUCED_REP1_trimmed_trimmed_fastqc.html",
                "RAP1_UNINDUCED_REP1_trimmed_trimmed_fastqc.zip",
                "RAP1_UNINDUCED_REP2",
                "RAP1_UNINDUCED_REP2.stats.txt",
                "RAP1_UNINDUCED_REP2_trimmed.fastq.gz_trimming_report.txt",
                "RAP1_UNINDUCED_REP2_trimmed_trimmed_fastqc.html",
                "RAP1_UNINDUCED_REP2_trimmed_trimmed_fastqc.zip",
                "WT_REP1",
                "WT_REP1.stats.txt",
                "WT_REP1_trimmed_1.fastq.gz_trimming_report.txt",
                "WT_REP1_trimmed_1_val_1_fastqc.html",
                "WT_REP1_trimmed_1_val_1_fastqc.zip",
                "WT_REP1_trimmed_2.fastq.gz_trimming_report.txt",
                "WT_REP1_trimmed_2_val_2_fastqc.html",
                "WT_REP1_trimmed_2_val_2_fastqc.zip",
                "WT_REP2",
                "WT_REP2.stats.txt",
                "WT_REP2_trimmed_1.fastq.gz_trimming_report.txt",
                "WT_REP2_trimmed_1_val_1_fastqc.html",
                "WT_REP2_trimmed_1_val_1_fastqc.zip",
                "WT_REP2_trimmed_2.fastq.gz_trimming_report.txt",
                "WT_REP2_trimmed_2_val_2_fastqc.html",
                "WT_REP2_trimmed_2_val_2_fastqc.zip",
                "ambig_info.tsv",
                "ambig_info.tsv",
                "ambig_info.tsv",
                "ambig_info.tsv",
                "ambig_info.tsv",
                "aux_info",
                "aux_info",
                "aux_info",
                "aux_info",
                "aux_info",
                "bbsplit",
                "cmd_info.json",
                "cmd_info.json",
                "cmd_info.json",
                "cmd_info.json",
                "cmd_info.json",
                "custom",
                "cutadapt_filtered_reads_plot-cnt.pdf",
                "cutadapt_filtered_reads_plot-cnt.png",
                "cutadapt_filtered_reads_plot-cnt.svg",
                "cutadapt_filtered_reads_plot-pct.pdf",
                "cutadapt_filtered_reads_plot-pct.png",
                "cutadapt_filtered_reads_plot-pct.svg",
                "cutadapt_filtered_reads_plot.txt",
                "cutadapt_trimmed_sequences_plot_3_Counts.pdf",
                "cutadapt_trimmed_sequences_plot_3_Counts.png",
                "cutadapt_trimmed_sequences_plot_3_Counts.svg",
                "cutadapt_trimmed_sequences_plot_3_Counts.txt",
                "cutadapt_trimmed_sequences_plot_3_Obs_Exp.pdf",
                "cutadapt_trimmed_sequences_plot_3_Obs_Exp.png",
                "cutadapt_trimmed_sequences_plot_3_Obs_Exp.svg",
                "cutadapt_trimmed_sequences_plot_3_Obs_Exp.txt",
                "expected_bias.gz",
                "expected_bias.gz",
                "expected_bias.gz",
                "expected_bias.gz",
                "expected_bias.gz",
                "fastqc",
                "fastqc-status-check-heatmap.pdf",
                "fastqc-status-check-heatmap.png",
                "fastqc-status-check-heatmap.svg",
                "fastqc-status-check-heatmap.txt",
                "fastqc_overrepresented_sequences_plot.pdf",
                "fastqc_overrepresented_sequences_plot.png",
                "fastqc_overrepresented_sequences_plot.svg",
                "fastqc_overrepresented_sequences_plot.txt",
                "fastqc_per_base_n_content_plot.pdf",
                "fastqc_per_base_n_content_plot.png",
                "fastqc_per_base_n_content_plot.svg",
                "fastqc_per_base_n_content_plot.txt",
                "fastqc_per_base_sequence_quality_plot.pdf",
                "fastqc_per_base_sequence_quality_plot.png",
                "fastqc_per_base_sequence_quality_plot.svg",
                "fastqc_per_base_sequence_quality_plot.txt",
                "fastqc_per_sequence_gc_content_plot_Counts.pdf",
                "fastqc_per_sequence_gc_content_plot_Counts.png",
                "fastqc_per_sequence_gc_content_plot_Counts.svg",
                "fastqc_per_sequence_gc_content_plot_Counts.txt",
                "fastqc_per_sequence_gc_content_plot_Percentages.pdf",
                "fastqc_per_sequence_gc_content_plot_Percentages.png",
                "fastqc_per_sequence_gc_content_plot_Percentages.svg",
                "fastqc_per_sequence_gc_content_plot_Percentages.txt",
                "fastqc_per_sequence_quality_scores_plot.pdf",
                "fastqc_per_sequence_quality_scores_plot.png",
                "fastqc_per_sequence_quality_scores_plot.svg",
                "fastqc_per_sequence_quality_scores_plot.txt",
                "fastqc_sequence_counts_plot-cnt.pdf",
                "fastqc_sequence_counts_plot-cnt.png",
                "fastqc_sequence_counts_plot-cnt.svg",
                "fastqc_sequence_counts_plot-pct.pdf",
                "fastqc_sequence_counts_plot-pct.png",
                "fastqc_sequence_counts_plot-pct.svg",
                "fastqc_sequence_counts_plot.txt",
                "fastqc_sequence_duplication_levels_plot.pdf",
                "fastqc_sequence_duplication_levels_plot.png",
                "fastqc_sequence_duplication_levels_plot.svg",
                "fastqc_sequence_duplication_levels_plot.txt",
                "fastqc_sequence_length_distribution_plot.pdf",
                "fastqc_sequence_length_distribution_plot.png",
                "fastqc_sequence_length_distribution_plot.svg",
                "fastqc_sequence_length_distribution_plot.txt",
                "fastqc_top_overrepresented_sequences_table.pdf",
                "fastqc_top_overrepresented_sequences_table.png",
                "fastqc_top_overrepresented_sequences_table.svg",
                "fastqc_top_overrepresented_sequences_table.txt",
                "fld.gz",
                "fld.gz",
                "fld.gz",
                "fld.gz",
                "fld.gz",
                "flenDist.txt",
                "flenDist.txt",
                "flenDist.txt",
                "flenDist.txt",
                "flenDist.txt",
                "general_stats_table.pdf",
                "general_stats_table.png",
                "general_stats_table.svg",
                "genome_gfp.fasta",
                "genome_gfp.gtf",
                "libParams",
                "libParams",
                "libParams",
                "libParams",
                "libParams",
                "lib_format_counts.json",
                "lib_format_counts.json",
                "lib_format_counts.json",
                "lib_format_counts.json",
                "lib_format_counts.json",
                "logs",
                "logs",
                "logs",
                "logs",
                "logs",
                "meta_info.json",
                "meta_info.json",
                "meta_info.json",
                "meta_info.json",
                "meta_info.json",
                "multiqc",
                "multiqc_citations.txt",
                "multiqc_cutadapt.txt",
                "multiqc_data.json",
                "multiqc_fastqc_fastqc_trimmed.txt",
                "multiqc_general_stats.txt",
                "multiqc_report.html",
                "multiqc_report_data",
                "multiqc_report_plots",
                "multiqc_salmon.txt",
                "multiqc_software_versions.txt",
                "multiqc_sources.txt",
                "nf_core_rnaseq_software_mqc_versions.yml",
                "observed_bias.gz",
                "observed_bias.gz",
                "observed_bias.gz",
                "observed_bias.gz",
                "observed_bias.gz",
                "observed_bias_3p.gz",
                "observed_bias_3p.gz",
                "observed_bias_3p.gz",
                "observed_bias_3p.gz",
                "observed_bias_3p.gz",
                "out",
                "pdf",
                "pipeline_info",
                "png",
                "quant.genes.sf",
                "quant.genes.sf",
                "quant.genes.sf",
                "quant.genes.sf",
                "quant.genes.sf",
                "quant.sf",
                "quant.sf",
                "quant.sf",
                "quant.sf",
                "quant.sf",
                "salmon",
                "salmon.merged.gene_counts.SummarizedExperiment.rds",
                "salmon.merged.gene_counts.tsv",
                "salmon.merged.gene_counts_length_scaled.SummarizedExperiment.rds",
                "salmon.merged.gene_counts_length_scaled.tsv",
                "salmon.merged.gene_counts_scaled.SummarizedExperiment.rds",
                "salmon.merged.gene_counts_scaled.tsv",
                "salmon.merged.gene_lengths.tsv",
                "salmon.merged.gene_tpm.tsv",
                "salmon.merged.transcript_counts.SummarizedExperiment.rds",
                "salmon.merged.transcript_counts.tsv",
                "salmon.merged.transcript_lengths.tsv",
                "salmon.merged.transcript_tpm.tsv",
                "salmon_plot.pdf",
                "salmon_plot.png",
                "salmon_plot.svg",
                "salmon_plot.txt",
                "salmon_quant.log",
                "salmon_quant.log",
                "salmon_quant.log",
                "salmon_quant.log",
                "salmon_quant.log",
                "svg",
                "trim",
                "trimgalore",
                "tx2gene.tsv"
            ],
            [
                "RAP1_IAA_30M_REP1.stats.txt:md5,a87292c6f7166096166a9015c2e63dac",
                "RAP1_IAA_30M_REP1_trimmed_1.fastq.gz_trimming_report.txt:md5,b64efca5b1f31cee893f00a00e43baea",
                "RAP1_IAA_30M_REP1_trimmed_2.fastq.gz_trimming_report.txt:md5,326657738d1f79eb9bc7735f44264b83",
                "RAP1_UNINDUCED_REP1.stats.txt:md5,781e49024150a8313779c574b58ad035",
                "RAP1_UNINDUCED_REP1_trimmed.fastq.gz_trimming_report.txt:md5,8265d3f3a4084617b6e1e27c2a9ed46c",
                "RAP1_UNINDUCED_REP2.stats.txt:md5,e455d0163bf2d988224b0c0092605213",
                "RAP1_UNINDUCED_REP2_trimmed.fastq.gz_trimming_report.txt:md5,ed85a0b7af3b2776c065a570d13eacff",
                "WT_REP1.stats.txt:md5,6eab6936804c1157d5e1c61c0562d7de",
                "WT_REP1_trimmed_1.fastq.gz_trimming_report.txt:md5,e88d0f0c93cf9f8377af01ed161389fe",
                "WT_REP1_trimmed_2.fastq.gz_trimming_report.txt:md5,4f43f1258e593219abbed5f25a087978",
                "WT_REP2.stats.txt:md5,002ddf6b828d34979652a40f05462843",
                "WT_REP2_trimmed_1.fastq.gz_trimming_report.txt:md5,12a83e73496de77bd9ac8042be365616",
                "WT_REP2_trimmed_2.fastq.gz_trimming_report.txt:md5,e73defcfa375bacacb3eede4b65b7b74",
                "ambig_info.tsv:md5,194f574e0586416155e3f33d42e2b167",
                "ambig_info.tsv:md5,6dcc2891ea572e9b8d1ba52cd434ab84",
                "ambig_info.tsv:md5,a26e3f936e65d7da66392603c2f91f6f",
                "ambig_info.tsv:md5,de973a4b22a4457217ae3dc04caf9401",
                "ambig_info.tsv:md5,45f252b4f0e11e6730cf0c29f800fdbb",
                "cmd_info.json:md5,c7ed0aaa5d6c7934ddbebfd29e4eb86d",
                "cmd_info.json:md5,9bed6e4dc5428d6f6297adcea29a6326",
                "cmd_info.json:md5,69ebfc2c7ca6b221a0a22fa1dc8c20ac",
                "cmd_info.json:md5,1215f0c20f87d3aef8553ef119e1e74c",
                "cmd_info.json:md5,621c6601aade5b1f2e3d6ca2fc71f636",
                "cutadapt_filtered_reads_plot.txt:md5,bf033e64e9d23bee85b6277f11c663f1",
                "cutadapt_trimmed_sequences_plot_3_Counts.txt:md5,13dfa866fd91dbb072689efe9aa83b1f",
                "cutadapt_trimmed_sequences_plot_3_Obs_Exp.txt:md5,07145dd8dd3db654859b18eb0389046c",
                "fastqc-status-check-heatmap.txt:md5,22a03548736b88b23be6bc0c9ef1b4a6",
                "fastqc_overrepresented_sequences_plot.txt:md5,4adfeacd3a3a6c7c808f121b24e6b247",
                "fastqc_per_base_n_content_plot.txt:md5,418610c1ce119cb786ad434db75d366e",
                "fastqc_per_base_sequence_quality_plot.txt:md5,bd22e06e41c096ad4f745d40fe96a1e5",
                "fastqc_per_sequence_gc_content_plot_Counts.txt:md5,004c60768ceb6197765154e3eaa37b7a",
                "fastqc_per_sequence_gc_content_plot_Percentages.txt:md5,95d29060b687f745288ad1ec47750037",
                "fastqc_per_sequence_quality_scores_plot.txt:md5,0f9834cc19f76dd5c87cf8cba7435a7c",
                "fastqc_sequence_counts_plot.txt:md5,3861354bbedfbde7ca36a72994f9425c",
                "fastqc_sequence_duplication_levels_plot.txt:md5,c73407d55fc532e864fa1dc8dbc12874",
                "fastqc_sequence_length_distribution_plot.txt:md5,6fe2c985606abad947bcca99b015ae33",
                "fastqc_top_overrepresented_sequences_table.txt:md5,cf612406874da4e582eaeaa956f7eb66",
                "flenDist.txt:md5,1a0b01789aabfb817e776e1c364fac85",
                "flenDist.txt:md5,2de170bdc9f6fd237d286429b292bb28",
                "flenDist.txt:md5,0aaae0a7f00b6f4037bc77e5f78b78d1",
                "flenDist.txt:md5,7e3ef392373f4de7bba7eb77cf93590e",
                "flenDist.txt:md5,2de170bdc9f6fd237d286429b292bb28",
                "lib_format_counts.json:md5,d46250bb3677d72feeefc435fe6395a6",
                "lib_format_counts.json:md5,7c562bf2f70e42f3a7292687dfd328c3",
                "lib_format_counts.json:md5,088fd51db07022ffde47033bbd029400",
                "lib_format_counts.json:md5,c24ffe28d70476b5ccdd8bc2d22c0ac1",
                "lib_format_counts.json:md5,f6d44c0221f7fd559f11a9afe04c9935",
                "meta_info.json:md5,e9132375893ce91db6947538e64dd43c",
                "meta_info.json:md5,5618fe266a5291cfef64e02f2de0547f",
                "meta_info.json:md5,9bfa0dba837c6ebd4003603548d8f4d1",
                "meta_info.json:md5,0d553bac54eda6f2a0e9ec847fae2bed",
                "meta_info.json:md5,d03a4f847bb05b6d512b48903981d95d",
                "multiqc_citations.txt:md5,f789abe663d4b4214f0ddeb413a7f150",
                "multiqc_cutadapt.txt:md5,aac9581a5670cb55edf564f3d6c1f9a7",
                "multiqc_fastqc_fastqc_trimmed.txt:md5,a3238f515e01d158d875d69968753804",
                "multiqc_general_stats.txt:md5,2c469bcca2d53528005d27eabc8abe09",
                "multiqc_salmon.txt:md5,c321e69be50c717245a31a3a1c99face",
                "salmon.merged.gene_counts.tsv:md5,c14cab7e15cfac73ec0602dc2c404551",
                "salmon.merged.gene_counts_length_scaled.tsv:md5,60c648d6bee7364d956c94e7b722c388",
                "salmon.merged.gene_counts_scaled.tsv:md5,7d28536335720847d2646c5565c2a2e8",
                "salmon.merged.gene_lengths.tsv:md5,dc106a20fd3f322d3122c7e718ad33ba",
                "salmon.merged.gene_tpm.tsv:md5,b6d982b0a0bedf3859bd0a950137c43e",
                "salmon.merged.transcript_counts.tsv:md5,ff0f5be09ca7a322672c0074ba35da17",
                "salmon.merged.transcript_lengths.tsv:md5,8d33a65a9c07bd59b1aab7d847be1e1a",
                "salmon.merged.transcript_tpm.tsv:md5,40350df578e9c6b42a4fd67d4fb9905b",
                "salmon_plot.txt:md5,afe2f326b018c7109a67f0cc86f23dc1",
                "tx2gene.tsv:md5,0e2418a69d2eba45097ebffc2f700bfe"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-13T11:52:59.260944"
    },
    "Params: --pseudo_aligner salmon --skip_qc --skip_alignment - stub": {
        "content": [
            {
                "BBMAP_BBSPLIT": {
                    "bbmap": 39.01
                },
                "CAT_FASTQ": {
                    "cat": 8.3
                },
                "CUSTOM_CATADDITIONALFASTA": {
                    "python": null
                },
                "CUSTOM_GETCHROMSIZES": {
                    "getchromsizes": 1.2
                },
                "GTF2BED": {
                    "perl": "5.26.2"
                },
                "GTF_FILTER": {
                    "python": "3.9.5"
                },
                "GUNZIP_ADDITIONAL_FASTA": {
                    "gunzip": 1.1
                },
                "GUNZIP_GTF": {
                    "gunzip": 1.1
                },
                "TRIMGALORE": {
                    "trimgalore": "0.6.7",
                    "cutadapt": 3.4
                },
                "UNTAR_SALMON_INDEX": {
                    "untar": 1.34
                },
                "Workflow": {
                    "nf-core/rnaseq": "v3.16.0dev"
                }
            },
            [
                "RAP1_IAA_30M_REP1_trimmed_1.fastq.gz_trimming_report.txt",
                "RAP1_IAA_30M_REP1_trimmed_2.fastq.gz_trimming_report.txt",
                "RAP1_UNINDUCED_REP1_trimmed.fastq.gz_trimming_report.txt",
                "RAP1_UNINDUCED_REP2_trimmed.fastq.gz_trimming_report.txt",
                "WT_REP1_trimmed_1.fastq.gz_trimming_report.txt",
                "WT_REP1_trimmed_2.fastq.gz_trimming_report.txt",
                "WT_REP2_trimmed_1.fastq.gz_trimming_report.txt",
                "WT_REP2_trimmed_2.fastq.gz_trimming_report.txt",
                "custom",
                "fastqc",
                "genome_transcriptome.fasta",
                "genome_transcriptome.gtf",
                "multiqc",
                "multiqc_data",
                "multiqc_plots",
                "multiqc_report.html",
                "nf_core_rnaseq_software_mqc_versions.yml",
                "out",
                "pipeline_info",
                "trim",
                "trimgalore"
            ],
            [
                "RAP1_IAA_30M_REP1_trimmed_1.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "RAP1_IAA_30M_REP1_trimmed_2.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "RAP1_UNINDUCED_REP1_trimmed.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "RAP1_UNINDUCED_REP2_trimmed.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "WT_REP1_trimmed_1.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "WT_REP1_trimmed_2.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "WT_REP2_trimmed_1.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "WT_REP2_trimmed_2.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-13T11:53:47.849117"
    }
}